clock/testing/fake_clock_test.go

```diff
@@ -194,3 +194,82 @@ func TestFakeStop(t *testing.T) {
 		t.Errorf("expected existing waiter to be cleaned up, but it is still present")
 	}
 }
+
+// This tests the pattern documented in the go docs here: https://golang.org/pkg/time/#Timer.Stop
+// This pattern is required to safely reset a timer, so should be common.
+// This also tests resetting the timer
+func TestFakeStopDrain(t *testing.T) {
+	tc := NewFakeClock(time.Now())
+	timer := tc.NewTimer(time.Second)
```

Comment:
This timer is used for a different test, right? If so, could you split into a different test?

Reply:
Indeed, sloppy copy pasting I think.